aws/config.go

```diff
@@ -759,3 +761,39 @@ func GetSupportedEC2Platforms(conn *ec2.EC2) ([]string, error) {
 
 	return platforms, nil
 }
+
+type wafv2CallFunc func() (interface{}, error)
+
+// Calls a WAFv2 action and retries on a retryable error
+func callWafv2WithRetry(f wafv2CallFunc) (interface{}, error) {
+	var out interface{}
+	err := resource.Retry(15*time.Minute, func() *resource.RetryError {
```

Comment:
Hi @pvanbuijtene 👋 Thank you for updating this! Our intention was to suggest marking the `wafv2.ErrCodeWAFInternalErrorException` and `wafv2.ErrCodeWAFTagOperationInternalErrorException` errors as retryable at the service client layer, since these seem like they should be globally retryable for any WAFv2 API call. This allows the AWS Go SDK to automatically perform the retries instead of our own retry logic via `resource.Retry()`. For example a few lines up from here:

https://github.com/terraform-providers/terraform-provider-aws/blob/07f8ab992c780c4911832234afbbf67feea2fe66/aws/config.go#L697-L710

Other errors like `wafv2.ErrCodeWAFAssociatedItemException` should keep the `resource.Retry()` logic in the resources directly, so it is clear what we are retrying on and for how long. 👍

We'd recommend not retrying `wafv2.ErrCodeWAFAssociatedItemException` for more than 5 minutes normally, since it may be unrecoverable due to a misconfiguration or lack of configuration. Also it'd be good to know why `wafv2.ErrCodeWAFTagOperationException` is included in this retry logic (potentially saved as a code comment if staying), since that seems like an unusual addition.

Reply:
@bflad totally missed that, the retry implementation is changed and looks cleaner now.

> Also it'd be good to know why wafv2.ErrCodeWAFTagOperationException is included in this retry logic

WAFv2 supports tag on create and these errors can be the result of a create action according to the documentation, I added this in a comment.

Reply:
@pvanbuijtene, apologies the confusion! my wording on updating the retry logic to the patterns used in the client was unclear. thank you for the quick refactor  🙏

Reply:
@anGie44 no apologies needed, I didn't notice the already present retry logic in that file.